src/lib/proof_system.ts

```diff
@@ -494,24 +599,39 @@ function picklesRuleFromFunction(
           : emptyWitness(type);
       } else if (arg.type === 'proof') {
         let Proof = proofArgs[arg.index];
-        let publicInput = getPublicInputType(Proof).fromFields(
-          previousInputs[arg.index]
-        );
-        let proofInstance: Proof<any>;
+        // split in input & output
+        // TODO: would be nice to either make Pickles return the right split or completely move
+        // getting the public input & outputs to the JS side, from `snarkContext`, like private inputs
```

Comment:
this TODO is addressed in a follow-up PR: https://github.com/o1-labs/snarkyjs/pull/876